test/lib/pwa.js

```diff
@@ -13,50 +13,61 @@
-let libLighthouse = require('../../lib/lighthouse');
-let db = require('../../lib/model-datastore');
-let cache = require('../../lib/data-cache');
+const fs = require('fs');
+const dataFetcher = require('../../lib/data-fetcher');
+const libPwa = require('../../lib/pwa');
+const libImages = require('../../lib/images');
+const libManifest = require('../../lib/manifest');
+const libLighthouse = require('../../lib/lighthouse');
+const db = require('../../lib/model-datastore');
+const cache = require('../../lib/data-cache');
 
-let Lighthouse = require('../../models/lighthouse');
-let Manifest = require('../../models/manifest');
-let Pwa = require('../../models/pwa');
+const Lighthouse = require('../../models/lighthouse');
+const Manifest = require('../../models/manifest');
+const Pwa = require('../../models/pwa');
 
-let simpleMock = require('simple-mock');
-let chai = require('chai');
-let chaiAsPromised = require('chai-as-promised');
+const simpleMock = require('simple-mock');
+const chai = require('chai');
+const chaiAsPromised = require('chai-as-promised');
 chai.use(chaiAsPromised);
 chai.should();
-let assert = require('chai').assert;
+const assert = require('chai').assert;
 
-const MANIFEST_URL = 'https://www.terra.com.br/manifest-br.json';
-const START_URL = 'https://www.terra.com.br/?utm_source=homescreen';
-const MANIFEST_DATA = './test/manifests/icon-url-with-parameter.json';
+const MANIFEST_URL = 'https://www.domain.com/manifest-br.json';
+const START_URL = 'https://www.domain.com/?utm_source=homescreen';
 const LIGHTHOUSE_JSON_EXAMPLE = './test/lib/lighthouse-example.json';
 
+/*eslint-disable */
+const MANIFEST_DATA =
+`{
+  "name": "Test",
+  "icons": [
+    {
+      "src": "img/launcher-icon.png?v2",
+      "sizes": "192x192",
+      "type": "image/png"
+    }
+  ],
+  "start_url": "https://www.example.com/?utm_source=homescreen"
+}`
+const MANIFEST_NO_ICON = `{"name": "Test", "description": "Manifest without icons", "start_url": "/"}`;
+const MANIFEST_INVALID_THEME_COLOR = `{"description": "Manifest with an invalid theme_color", "theme_color": ""}`;
+/*eslint-enable */
+
 describe('lib.pwa', () => {
-  let manifest;
-  let pwa;
-  let lighthouse;
-  before(done => {
-    dataFetcher.readFile(MANIFEST_DATA)
-      .then(jsonString => {
-        manifest = new Manifest(MANIFEST_URL, JSON.parse(jsonString));
-        pwa = new Pwa(MANIFEST_URL, manifest);
-        pwa.id = '123456789';
-        dataFetcher.readFile(LIGHTHOUSE_JSON_EXAMPLE)
-          .then(data => {
-            lighthouse = new Lighthouse('123456789', 'www.domain.com', JSON.parse(data));
-            done();
-          });
-      });
-  });
+  const manifest = new Manifest(MANIFEST_URL, JSON.parse(MANIFEST_DATA));
+  const pwa = new Pwa(MANIFEST_URL, manifest);
+  pwa.id = '123456789';
+  const manifestNoIcon = new Manifest(MANIFEST_URL, JSON.parse(MANIFEST_NO_ICON));
+  const pwaNoIcon = new Pwa(MANIFEST_URL, manifestNoIcon);
```

Comment:
I've created a helper method for instantiating the PWAs [here](https://github.com/GoogleChrome/gulliver/blob/master/test/models/pwa.js#L129). We could use this everytime we create a new pwa in a test. The benefit is that we only need to touch the helper method if for example the PWA constructor API changes.